frontend/server/src/Validators.php

```diff
@@ -760,4 +763,112 @@ public static function validateBadgeExists(
+            throw new \OmegaUp\Exceptions\InvalidParameterException(
+                'parameterInvalidValidZipUpload',
+                'zipFile'
+            );
+        }
+    }
+
+    /**
+     * Check the size of the ZIP file
+     *
+     * @param string $filePath
+     * @throws \OmegaUp\Exceptions\InvalidParameterException
+     */
+    public static function validateZipFileSize(string $filePath): void {
+        $fileSize = filesize($filePath);
+
+        if ($fileSize === false) {
+            throw new \OmegaUp\Exceptions\InvalidParameterException(
+                'parameterInvalidFileSize',
+                'zipFile'
+            );
+        }
+
+        if ($fileSize > self::ZIP_MAX_FILE_SIZE_BYTES) {
+            throw new \OmegaUp\Exceptions\InvalidParameterException(
+                'parameterStringTooLong',
+                'zipFile',
+                ['max_length' => strval(self::ZIP_MAX_FILE_SIZE_BYTES / (1024 * 1024))]
+            );
+        }
+
+        if ($fileSize === 0) {
+            throw new \OmegaUp\Exceptions\InvalidParameterException(
+                'parameterEmpty',
+                'zipFile'
+            );
+        }
+    }
+
+    /**
+     * Check that it is a valid ZIP file
+     *
+     * @param string $filePath
+     * @throws \OmegaUp\Exceptions\InvalidParameterException
+     */
+    public static function validateZipIntegrity(string $filePath): void {
+        $zip = new \ZipArchive();
+
+        if ($zip->open($filePath) !== TRUE) {
+            throw new \OmegaUp\Exceptions\InvalidParameterException(
+                'parameterInvalidZipIntegrity',
+                'zipFile'
+            );
+        }
+
+        $zip->close();
+    }
+
+    /**
+     * Check the path of a file within the ZIP (prevents path traversal)
+     *
+     * @param string $filePath
+     * @throws \OmegaUp\Exceptions\InvalidParameterException
+     */
+    public static function validateZipFilePath(string $filePath): void {
+        foreach (self::ZIP_FORBIDDEN_PATH_CHARS as $char) {
+            if (strpos($filePath, $char) !== false) {
+                throw new \OmegaUp\Exceptions\InvalidParameterException(
+                    'parameterInvalidZipPath',
+                    'zipFile'
+                );
+            }
+        }
+    }
+
+    /**
+     * Check the file extension and case name
+     *
+     * @param string $filename
+     * @param string $extension
+     * @throws \OmegaUp\Exceptions\InvalidParameterException
+     */
+    public static function validateZipCaseFileName(string $filename, string $extension): void {
+        if (!in_array($extension, self::ZIP_ALLOWED_CASE_EXTENSIONS)) {
+            throw new \OmegaUp\Exceptions\InvalidParameterException(
+                'parameterInvalidExtension',
+                'zipFile'
+            );
+        }
+
+        // Check that the name does not contain more than one period.
+        $parts = explode('.', $filename);
+        if (count($parts) > 2) {
```

Comment:
Según el comentario, más de un punto debería arrojar el error:
```suggestion
        if (count($parts) >= 2) {
```
Si la condición estaba bien, entonces lo que hay que arreglar es el comentario

Reply:
El comentario lo expresé de esa manera porque explode('.', $filename) divide el nombre del archivo por cada punto, y `count($parts) > 2 `detecta correctamente si hay más de un punto, lo cual es el comportamiento esperado. ¿Crees que sería mejor cambiar el comentario por uno más explicativo?

Reply:
No, así está bien. Creo que era yo el que me estaba confundiendo